firebase-dynamic-links/ktx/src/main/kotlin/com/google/firebase/dynamiclinks/ktx/FirebaseDynamicLinks.kt

```diff
@@ -117,6 +120,50 @@ fun FirebaseDynamicLinks.shortLinkAsync(suffix: Int, init: DynamicLink.Builder.(
     return builder.buildShortDynamicLink(suffix)
 }
 
+/**
+ * Returns the [PendingDynamicLinkData] present in this [Intent], using the
+ * [FirebaseDynamicLinks] instance of the given [FirebaseApp].
+ */
+fun Intent.getFirebaseDynamicLink(firebaseApp: FirebaseApp): Task<PendingDynamicLinkData> {
```

Comment:
I think this may be overkill, today you can do this with `Firebase.dynamicLinks(app).getDynamicLink(intent)`,

same for the Uri extensions.

wdyt?

Reply:
@vkryachko The initial motivation for these functions was to improve readability the same way Android KTX does (e.g. `"https://firebase.google.com".toUri()` instead of `Uri.parse("https://firebase.google.com")`)

I then realized developers may want to use their own `FirebaseApp` instance, and passing it as argument seemed like the only way. But I now realize it totally defeats the initial "improved readability" purpose.

Maybe we should drop the extensions that take `FirebaseApp` as argument, but keep the ones with no arguments? 
What do you say?

Reply:
hm, it's hard to objectively judge the usefulness of `"https://firebase.google.com".toUri()`, I guess it is useful if it's included in android ktx.

One difference it has from your proposal though is that `intent.getDynamicLink()` returns a task indicating not a trivial transformation but some IO heave work. In this regard I would be hesitant to optimize for conciseness.

Additionally, in the case of the default app, `Firebase.dynamicLinks.getDynamicLink(intent)` does not look much less readable to me.

Reply:
@vkryachko I totally understand your points. I'll remove these extensions and keep the destructuring declarations.